### ISSUE TITLE
Duplicate guest checks

### ISSUE THREAD

**Issue by connojd** [Bareflank/hypervisor#225]
``` cpp
vmcs_intel_x64::check_guest_valid_activity_state
```

 and 

``` cpp
vmcs_intel_x64::check_guest_activity_state_not_hlt_when_dpl_not_0
```

are called twice in `vmcs_intel_x64::checks_on_guest_non_register_state`.

**Comment by rianquinn** [comment]
Oppsss

**Comment by connojd** [comment]
Fixed in #343
